patches/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java.patch

```diff
@@ -0,0 +1,14 @@
+--- ../src-base/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
++++ ../src-work/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
+@@ -184,9 +184,9 @@
+ 
+                 if (!flag)
+                 {
+-                    Block block = p_178270_1_.func_180495_p(blockpos).func_177230_c();
++                    IBlockState state = p_178270_1_.func_180495_p(blockpos);
+ 
+-                    if (block == Blocks.field_150359_w || block == Blocks.field_150399_cn)
++                    if (state.func_193401_d(p_178270_1_, blockpos, EnumFacing.field_82609_l[i1+2].func_176734_d()) == net.minecraft.block.state.BlockFaceShape.SOLID)
```

Comment:
I think `EnumFacing.VALUES[i1+2]` should be `EnumFacing.getHorizontal(i1)`

Reply:
That doesn't work because `VALUES` and `HORIZONTALS` return values in different orders. (`getHorizontal()` just returns the values from `HORIZONTALS`)

VALUES[2] -> NORTH
VALUES[3] -> SOUTH
VALUES[4] -> WEST
VALUES[5] -> EAST

HORIZONTALS[0] -> SOUTH
HORIZONTALS[1] -> WEST
HORIZONTALS[2] -> NORTH
HORIZONTALS[3] -> EAST

`BlockFluidRenderer::renderFluid` checks the adjacent blocks in the same order as `VALUES`, so if I used `EnumFacing.getHorizontal(i1)` it would check the wrong side of the block, it wouldn't check the side that's against water.

Reply:
Probably would make sense to make `HORIZONTALS` follow the same order as `VALUES`. Then it would be possible to use getHorizontal(i1) instead of `VALUES[i1+2]`. Although I'm not sure if it would break something that depends on it being in that specific order?

Reply:
Yuck, ok. Nothing we can do about that, yeah.